Don't cancel pending visualization's upserts

hubertp opened the pull request:
### Pull Request Description

Uniqueness check of `UpsertVisualizationJob` only involved expressionId. Apparently now GUI sends mutliple visualizations for the same expressions and expects all of them to exist. Since previously we would cancel duplicate jobs, this was problematic.

This change makes sure that uniqueness also takes into account visualization id. Fixed a few logs that were not passing arguments properly.

Closes #8801 

### Important Notes

I have not noticed any more problems with loading visualizations so the issue appears to be resolved with this change.
Added a unit test case that would previously fail due to cancellation of a job that upserts visualization.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

hubertp committed:
Don't cancel pending visualization's upserts

Uniqueness check of `UpsertVisualizationJob` only involved expressionId.
Apparently now GUI sends mutliple visualizations for the same
expressions and expects all of them to exist. Since previously we would
cancel duplicate jobs, this was problematic.

This change makes sure that uniqueness also takes into account
visualization id. Fixed a few logs that were not passing arguments
properly.

hubertp committed:
Add test illustrating the problem

Added a problematic case where visualization upserts would get cancelled
because uniqueness took into account only expression id but not
visualization id.
The test is a bit tricky to setup because it relies on
a) multiple attach visualization requests
b) background job trigger cleaning up duplicate jobs (here triggered via
edit)

hubertp commented on engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala:
```diff
@@ -53,7 +53,7 @@ class UpsertVisualizationJob(
     that match {
       case that: UpsertVisualizationJob =>
-        this.expressionId == that.expressionId
+        this.expressionId == that.expressionId && this.visualizationId == that.visualizationId
```
This was the root of the problem.

JaroslavTulach replied:
Might deserve a simple unit test comparing two job instances for equality. The `RuntimeVisualizationsTest` may contain such check, but it is very hidden in the whole logic of the test.

hubertp replied:
Added

JaroslavTulach replied:
I see 1d8cd25e9315e3348d8700778105001903eb3b8d, thanks.

hubertp commented on engine/runtime/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala:
```diff
@@ -2058,512 +1169,555 @@ class RuntimeVisualizationsTest
+
+    attachVisualizationResponses.filter(
+      _.payload.isInstanceOf[Api.VisualizationAttached]
+    ) shouldEqual List(
```
This test would previously fail.

JaroslavTulach reviewed (approved):
Mostly process oriented approval. I didn't try to understand the changes in `RuntimeVisualizationsTest`.

hubertp committed:
Add unit test for uniqueness

hubertp committed:
Merge branch 'develop' into wip/hubert/8801-upsert-visualization-unique

hubertp committed:
typo after merge

hubertp committed:
formatting

hubertp committed:
make the test more resilient

hubertp committed:
Add some logging to help with CI debugging

hubertp commented:
Based on the last failures I basically think that `runtime/test` does not attempt to recompile tests in `runtime-instrument-common` on which it indirectly depends on. Then no wonder it does **not** have the required delays to simulate the problem.

hubertp committed:
Workaround cyclic dependency between runtime projects

`runtime-instrument-common` depends on `runtime` but `runtime/test`
depends indirectly on `runtime-instrument-commont`. We somehow managed
to deal with it for modules, but separate compilation breaks everything
in CI. This workaround will make sure that CI passes :fingers-crossed:.

hubertp committed:
Make test more resilient to randomness

hubertp committed:
Remove circular dependency for runtime

The circular dependency between `runtime` and
`runtime-instrument-common` is the root of all evil. There is no way
around until now, at the cost of some duplication.
Should fix issues in incremental compilation when changing between
branches.

hubertp committed:
formatting

hubertp committed:
Disable flaky test

Working on a fix in a separate PR.
